src/scheduler/node_info.c

```diff
@@ -4849,6 +4852,12 @@ parse_execvnode(char *execvnode, server_info *sinfo, selspec *sel)
 			if (simplespec != NULL) {
 				int ret;
 
+				if (hp > 0) /* simplespec starts with '(' but doesn't end with ')' */
+					in_superchunk = 1;
+				else if (hp < 0) /* simplespec ends with ')' but does not start with '(' */
+					in_superchunk = 0;
```

Comment:
Do we need to change anything if hp == 0?

Reply:
No, I added a comment for what hp == 0 means just for completeness.  hp == 0 means either (chunk) or chunk without ().  (chunk) is not a superchunk, so we don't need to change anything.  chunk without () is the middle of a superchunk, so we don't need to change anything either.